connectors/sources/microsoft_teams.py

```diff
@@ -281,6 +720,475 @@ async def ping(self):
             self._logger.exception("Error while connecting to Microsoft Teams")
             raise
 
+    def format_doc(self, item, document_type, **kwargs):
+        document = {}
+        for elasticsearch_field, sharepoint_field in kwargs["document"].items():
+            document[elasticsearch_field] = sharepoint_field
+        self.map_document_with_schema(
+            document=document, item=item, document_type=document_type
+        )
+        return document
+
+    def format_user_chat_meeting_recording(self, item, url):
+        document = {"type": UserEndpointName.MEETING_RECORDING.value}
+        document.update(
+            {
+                "_id": item.get("eventDetail", {}).get("callId"),
+                "title": item.get("eventDetail", {}).get("callRecordingDisplayName"),
+                "url": url,
+                "_timestamp": item.get("lastModifiedDateTime"),
+            }
+        )
+        return document
+
+    def get_calendar_detail(self, calendar):
+        body = ""
+        organizer = calendar.get("organizer", {}).get("emailAddress").get("name")
+        if calendar.get("recurrence"):
+            recurrence_range = calendar.get("recurrence", {}).get("range")
+            pattern = calendar.get("recurrence", {}).get("pattern")
```

Comment:
`calendar.get("recurrence")` is called a lot, it could be a variable. You wouldn't have to worry about safe-calling it into a dict this way either.

```suggestion
        calendar_recurrence = calendar.get("recurrence")
        if calendar_recurrence:
            recurrence_range = calendar_recurrence.get("range")
            pattern = calendar_recurrence.get("pattern")
```